From 25fc4c90a03679bdb7fd1186a15b19f5d5b7e38d Mon Sep 17 00:00:00 2001
From: David Daney <david.daney@cavium.com>
Date: Fri, 21 Sep 2012 14:02:46 -0700
Subject: [PATCH 166/337] MIPS: OCTEON: Change load address to waste less
 memory.

Based On SDK 3.0.0-482

"Modern" octeon bootloaders allow us to load the image at 0x100000.
Moving there from 0x1100000 makes 16MB more memory usable.

Signed-off-by: David Daney <david.daney@cavium.com>
Signed-off-by: Corey Minyard <cminyard@mvista.com>
Signed-off-by: Jack Tan <jack.tan@windriver.com>
---
 arch/mips/Makefile               |    4 ----
 arch/mips/cavium-octeon/Platform |    8 +++++++-
 2 files changed, 7 insertions(+), 5 deletions(-)

diff --git a/arch/mips/Makefile b/arch/mips/Makefile
index 0357326..522b472 100644
--- a/arch/mips/Makefile
+++ b/arch/mips/Makefile
@@ -209,10 +209,6 @@ include $(srctree)/arch/mips/Kbuild.platforms
 cflags-y			+= -I$(srctree)/arch/mips/include/asm/mach-generic
 drivers-$(CONFIG_PCI)		+= arch/mips/pci/
 
-ifdef CONFIG_MAPPED_KERNEL
-  load-y = 0xffffffffc0000000
-endif
-
 #
 # Automatically detect the build format. By default we choose
 # the elf format according to the load address.
diff --git a/arch/mips/cavium-octeon/Platform b/arch/mips/cavium-octeon/Platform
index ef9871a..27e63ca 100644
--- a/arch/mips/cavium-octeon/Platform
+++ b/arch/mips/cavium-octeon/Platform
@@ -4,4 +4,10 @@
 platform-$(CONFIG_CPU_CAVIUM_OCTEON)	+= cavium-octeon/
 cflags-$(CONFIG_CPU_CAVIUM_OCTEON)	+=				\
 		-I$(srctree)/arch/mips/include/asm/mach-cavium-octeon
-load-$(CONFIG_CPU_CAVIUM_OCTEON)	+= 0xffffffff81100000
+
+ifdef CONFIG_MAPPED_KERNEL
+  load-$(CONFIG_CPU_CAVIUM_OCTEON)	= 0xffffffffc0000000
+else
+  load-$(CONFIG_CPU_CAVIUM_OCTEON)	= 0xffffffff80100000
+endif
+
-- 
1.7.5.4

